primitives/state-machine/src/changes_trie/build.rs

```diff
@@ -105,20 +105,18 @@ fn prepare_extrinsics_input<'a, B, H, Number>(
 		H: Hasher + 'a,
 		Number: BlockNumber,
 {
-
-	let mut children_info = BTreeSet::<ChildInfo>::new();
 	let mut children_result = BTreeMap::new();
-	for (_storage_key, (_map, child_info)) in changes.prospective.children_default.iter()
-		.chain(changes.committed.children_default.iter()) {
-		children_info.insert(child_info.clone());
-	}
-	for child_info in children_info {
+
+	for child_info in changes.child_infos() {
```

Comment:
This is wrong. Before we filtered duplicates by using the `BTreeSet` and now we don't filter the duplicates. It may not change the output (as far as I can see and understand the code on a short review), however it will probably be faster to filter duplicates before iterating them.

Reply:
Added de duplication in the `fn child_infos`.